subsys/testsuite/ztest/include/ztest_assert.h

```diff
@@ -50,21 +50,29 @@ static inline bool z_zassert(bool cond,
 			    int line, const char *func,
 			    const char *msg, ...)
 {
+	static char formatted_msg[512]; /* TODO: is this the right size? */
+
 	if (cond == false) {
 		va_list vargs;
+		int ret;
 
 		va_start(vargs, msg);
-		PRINT("\n    Assertion failed at %s:%d: %s: %s\n",
-		      ztest_relative_filename(file), line, func, default_msg);
-		vprintk(msg, vargs);
-		printk("\n");
+		ZTEST_ERR("\n    Assertion failed at %s:%d: %s: %s\n",
+			  ztest_relative_filename(file), line, func, default_msg);
+
+		ret = vsnprintf(formatted_msg, sizeof(formatted_msg), msg, vargs);
+		if (ret >= sizeof(formatted_msg)) {
+			ZTEST_ERR("The assertion message has been truncated!\n");
+		}
+		ZTEST_ERR("%s\n", formatted_msg);
```

Comment:
Can you use `Z_LOG_VA`?

Alternatively (if not), I think your best bet is to actually turn `z_zassert` into a macro, then you can just pass `msg` and `__VA_ARGS__` to `ZTEST_ERR`